diff --git a/core/modules/views/src/Plugin/views/display/EntityReference.php b/core/modules/views/src/Plugin/views/display/EntityReference.php
index 47c372b..8c1ebb9 100644
--- a/core/modules/views/src/Plugin/views/display/EntityReference.php
+++ b/core/modules/views/src/Plugin/views/display/EntityReference.php
@@ -192,7 +192,9 @@ class EntityReference extends DisplayPluginBase {
       $this->view->query->addWhere(0, $id_table . '.' . $id_field, $options['ids'], 'IN');
     }

-    $this->view->setItemsPerPage($options['limit']);
+    if(isset($options['limit'])) {
+      $this->view->setItemsPerPage($options['limit']);
+    }
   }
