diff --git a/core/modules/action/action.module b/core/modules/action/action.module
index 0e4229debc..f3ac61db22 100644
--- a/core/modules/action/action.module
+++ b/core/modules/action/action.module
@@ -61,12 +61,16 @@ function action_migration_plugins_alter(array &$migrations) {
   foreach ($migrations as $migration_id => $migration) {
     // Add Actions plugins in actions module.
     /** @var \Drupal\migrate\Plugin\migrate\source\SqlBase $source_plugin */
-    $source_plugin = \Drupal::service('plugin.manager.migration')
-      ->createStubMigration($migration)
-      ->getSourcePlugin();
-    if (is_a($source_plugin, Action::class) && isset($migration['process']['plugin'])) {
-      $migrations[$migration_id]['process']['plugin'][0]['map']['comment_unpublish_by_keyword_action'] = 'comment_unpublish_by_keyword_action';
-      $migrations[$migration_id]['process']['plugin'][0]['map']['node_unpublish_by_keyword_action'] = 'node_unpublish_by_keyword_action';
+    try {
+      $source_plugin = \Drupal::service('plugin.manager.migration')
+        ->createStubMigration($migration)
+        ->getSourcePlugin();
+      if (is_a($source_plugin, Action::class) && isset($migration['process']['plugin'])) {
+        $migrations[$migration_id]['process']['plugin'][0]['map']['comment_unpublish_by_keyword_action'] = 'comment_unpublish_by_keyword_action';
+        $migrations[$migration_id]['process']['plugin'][0]['map']['node_unpublish_by_keyword_action'] = 'node_unpublish_by_keyword_action';
+      }
+    } catch (Exception $e) {
+      // Do nothing. This hack added to allow the XML upload to work.
     }
   }
 }
